core/remotes/docker/config/hosts.go

```diff
@@ -267,7 +235,57 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 
 		return rhosts, nil
 	}
+}
+
+// tlsConfigFromHost updates `tlsConfig` with the TLS configuration from `host`.
+// Use `host` as a pointer to avoid struct copy.
+func tlsConfigFromHost(tlsConfig *tls.Config, host *hostConfig) error {
```

Comment:
This is moved from old line 182-229. The `ConfigureHosts` func is a bit too long IMO. I feel moving this out of it makes its main logic more clear.